File: killrvideo-service-videocatalog/src/main/java/com/killrvideo/service/video/dto/LatestVideosPage.java
package com.killrvideo.service.video.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.checkerframework.checker.units.qual.A;

import java.util.ArrayList;
import java.util.List;

/**
 * Latest page.
 *
 * @author DataStax Developer Advocates team.
 */
@Getter @Setter @NoArgsConstructor @AllArgsConstructor
public class LatestVideosPage {
    /** List of Previews. */
    private List<LatestVideo> listOfPreview = new ArrayList<>();
    
    /** Flag if paging state. */
    private String cassandraPagingState = "";
    
    /** Use to return for query. */
    private String nextPageState = "";

    public int getResultSize() {
        return getListOfPreview().size();
    }
}

File: killrvideo-service-videocatalog/src/main/kotlin/com/killrvideo/service/video/dto/LatestVideosPage.kt
package com.killrvideo.service.video.dto

/**
 * Latest page.
 *
 * @author DataStax Developer Advocates team.
 */
data class LatestVideosPage(
    /** List of Previews.  */
    val listOfPreview: MutableList<LatestVideo>,
    /** Flag if paging state.  */
    var cassandraPagingState: String? = null,
    /** Use to return for query.  */
    var nextPageState: String? = null
) {
    constructor(): this(
        listOfPreview = mutableListOf()
    )

    val resultSize: Int
        get() = listOfPreview.size
}
